.github/workflows/changed_file_trigger_examples_check.yml

```diff
@@ -0,0 +1,128 @@
+name: Test Example
+on:
+  pull_request:
+  schedule:
+    # run at 00:00 of every Sunday(singapore time) so here is UTC time Saturday 16:00
+    - cron:  '0 16 * * 6'
+
+jobs:
+# This is for changed files detect and check.
+  detect-changed-files:
+    runs-on: ubuntu-latest
+    outputs:
+      changedFiles: ${{ steps.changed-files.outputs.all_changed_files }}
+    name: Check out all files
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - name: Get all changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v35
+        with:
+          since_last_remote_commit: true  # Using this can trigger action each time a PR is submitted.
+
+  prepare-changed-file-matrix:
```

Comment:
`detect-changed-files` and `prepare-changed-file-matrix` can be merged into one step so there is no need to checkout twice.